app/models/user.rb

```diff
@@ -3,11 +3,15 @@ class User < ActiveRecord::Base
 
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  scope :mentors, -> { where(mentor: true) }
+  scope :mentorees, -> { where(mentor: false) }
+
   def to_s
     user_name
   end
 
   def unify
-    self.find_related_skills
+    self.mentor ? self.find_related_skills.mentorees : self.find_related_skills
```

Comment:
You could remove all these `self` and it would still work fine. But that's just a personal choice.